packages/core/contracts/oracle/implementation/Staker.sol

```diff
@@ -196,31 +216,57 @@ abstract contract Staker is StakerInterface, Ownable, Lockable {
      * @notice Send accumulated rewards to the voter. Note that these rewards do not include slashing balance changes.
      * @return uint256 the amount of tokens sent to the voter.
      */
-    function withdrawRewards() public override nonReentrant() returns (uint256) {
-        _updateTrackers(msg.sender);
-        VoterStake storage voterStake = voterStakes[msg.sender];
+    function withdrawRewards() public returns (uint256) {
+        return _withdrawRewards(msg.sender, msg.sender);
+    }
+
+    function _withdrawRewards(address voter, address recipient) internal returns (uint256) {
+        _updateTrackers(voter);
+        VoterStake storage voterStake = voterStakes[voter];
 
         uint256 tokensToMint = voterStake.outstandingRewards;
         if (tokensToMint > 0) {
             voterStake.outstandingRewards = 0;
-            require(votingToken.mint(msg.sender, tokensToMint), "Voting token issuance failed");
-            emit WithdrawnRewards(msg.sender, tokensToMint);
+            require(votingToken.mint(recipient, tokensToMint), "Voting token issuance failed");
+            emit WithdrawnRewards(voter, msg.sender, tokensToMint);
         }
         return tokensToMint;
     }
 
     /**
-     * @notice Stake accumulated rewards. This is just a convenience method that combines withdraw with stake in the
-     * same transaction.
-     * @dev this method requires that the user has approved this contract.
-     * @return uint256 the amount of tokens that the user is staking.
+     * @notice Stake accumulated rewards. This is merely a convenience mechanism that combines the voter's withdrawal and stake
+     *  in the same transaction if requested by a delegate or the voter himself.
+     * @dev this method requires that the msg.sender(voter or delegate) has approved this contract.
+     * @dev The rewarded tokens simply pass through the delegate's wallet before being staked on the voter's behalf.
```

Comment:
is this comment accurate? my understanding of the flow you've got now is that you simply mint the tokens directly to the Staker contract and they don't actually leave the contract; they stay here and we simply increment the voters state.

Reply:
Agree. This flow has been changed, so this should say that the tokens are minted directly to the contract.